haystack/propagator.py

```diff
@@ -0,0 +1,12 @@
+from abc import ABC, abstractmethod
+
+
+class Propagator(ABC):
+
+    @abstractmethod
+    def inject(self, span_context, carrier):
+        pass
```

Comment:
Can we throw an exception with a custom message? It will make it clear for any person that extends it.

Reply:
now raising NotImplemented Exception